session/session.go

```diff
@@ -1385,6 +1386,12 @@ func (s *session) ParseWithParams(ctx context.Context, sql string, args ...inter
 	for _, warn := range warns {
 		s.sessionVars.StmtCtx.AppendWarning(util.SyntaxWarn(warn))
 	}
+	if config.TopSQLEnabled() {
+		normalized, digest := parser.NormalizeDigest(sql)
+		if digest != nil {
+			tracecpu.SetGoroutineLabelsWithSQL(ctx, normalized, digest.String())
```

Comment:
it seems this method be called many places, what behavior when it be called multiple time with the same ctx ❓ 

for example https://github.com/pingcap/tidb/blob/3a1dd7a6651802795a59188c38108ba085cc101b/statistics/handle/gc.go#L57-L64

Reply:
Nice catch, I need to unset the pprof label for  `execRestrictedSQL`.

It is ok to call `SetGoroutineLabelsWithSQL ` multiple times with the same ctx and same label.

Reply:
There is too many places need to reset the goroutine label. And this PR already too large, so I add a `Fixme` comment to fix this problem in the next feature. It only affects internal SQL CPU time precision。